knx/dpt/formats.go

```diff
@@ -30,6 +30,40 @@ func unpackB1(data []byte, b *bool) error {
 	return nil
 }
 
+func packB4(bs [4]bool) byte {
+	var b byte
+	b = 0
+	if bs[0] {
+		b |= 1 << 0
+	}
+	if bs[1] {
+		b |= 1 << 1
+	}
+	if bs[2] {
+		b |= 1 << 2
+	}
+	if bs[3] {
+		b |= 1 << 3
+	}
+
+	return byte(b)
+
+}
+
+func unpackB4(data byte, b0 *bool, b1 *bool, b2 *bool, b3 *bool) error {
+
+	if uint8(data) > 15 {
+		return ErrInvalidLength
```

Comment:
This error isn't accurate. I think an error besides `ErrInvalidLength` which indicates that reserved bits of a bitmap have been populated would fit better.

```go
var ErrBadReservedBits = errors.New("reserved bits in the input data have been populated")
```

Reply:
Yes sure. Will fix it.